### ISSUE TITLE
Question about your code

### ISSUE THREAD

**Issue by zhangzp9970** [thuml/Universal-Domain-Adaptation#5]
Hello, after reading the paper and the code, I have a few questions about them. Wish you could answer them for me :)
The UAN model proposed in the paper looks like this

![TIM图片20200214122712](https://user-images.githubusercontent.com/56378403/74501610-ab077b80-4f25-11ea-8828-d366cf6eec32.png).
 In my view, the feature z extracted from F is loaded into G, D, D' separately. However, the model in the code is defined as follow:
```python
def forward(self, x):
        f = self.feature_extractor(x)
        f, _, __, y = self.classifier(f)
        d = self.discriminator(_)
        d_0 = self.discriminator_separate(_)
        return y, d, d_0
```
The feature f goes into classifier first but the output of the first layer of the classifier goes into the discriminator? Why?

Besides, the training of the network can be seen as a minimax game of equation (4). However, the code just added all the loss up `loss = ce + adv_loss + adv_loss_separate`. How to understand this?

Thanks

**Comment by nmakes** [comment]
@zhangzp9970 : There is a Gradient Reversal Layer in the discriminator (Line 157, [Here](https://github.com/thuml/Universal-Domain-Adaptation/blob/master/net.py)), so there will be an adversarial loss between F and D.

I'm not sure about inputs to the discriminators though.

**Comment by zhangzp9970** [comment]
@nmakes Thanks for answer! I have found that in paper [Domain-Adversarial Training of Neural Networks](https://arxiv.org/abs/1505.07818) the author mentioned that The adaptation architecture is identical to Tzeng et al. (2014) which has "a 2-layer domain classifier (x->1024->1024->2) is attached to the 256-dimensional bottleneck of fc7." 
And in the paper [Tzeng et,al. Deep domainconfusion: Maximizing for domain invariance](http://arxiv.org/abs/1412.3474) the author chose different layers of AlexNet(fc6,fc7,fc8) and different dimensions of the bottleneck(from 64 to 4096). He found that fc7 layer and 256 dimension would be a reasonable choice, since it got the lowest MMD value. So, I think this is why features of 256 dimension is put into the adversarial network.
